site/app/models/Notification.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace app\models;
+
+use app\libraries\Core;
+
+/**
+ * Class Notification
+ *
+ * @method string getComponent()
+ * @method string getCurrentUser()
+ * @method string getNotifySource()
+ * @method string getNotifyTarget()
+ * @method string getNotifyContent()
+ * @method string getNotifyMetadata()
+ * @method bool getNotifyNotToSource()
+ */
+class Notification extends AbstractModel {
+
+    /** @property @var string Type of notification */
+    protected $component;
+    /** @property @var string Current logged in user */
+    protected $current_user;
+
+    /** @property @var string Notification source user (can be null) */
+    protected $notify_source;
+    /** @property @var string Notification target user(s) (null implies all users) */
+    protected $notify_target;
+    /** @property @var string Notification text content */
+    protected $notify_content;
+    /** @property @var string Notification information about redirection link */
+    protected $notify_metadata;
+    /** @property @var string Should $notify_source be ignored from $notify_target */
+    protected $notify_not_to_source;
+
+    /**
+     * Notifications constructor.
+     *
+     * @param Core  $core
+     * @param array $details
+     */
+    public function __construct(Core $core, $details=array()) {
+        parent::__construct($core);
+        if (count($details) == 0) {
+            return;
+        }
+        $this->setNotifyNotToSource(true);
+        $this->setCurrentUser($this->core->getUser()->getId());
+        $this->setComponent($details['component']);
+        switch ($this->getComponent()) {
+            case 'forum':
+                $this->handleForum($details);
+                break;
+            default:
+                // Prevent notification to be pushed in database
+                $this->setComponent("invalid");
+                break;
+        }
+    }
+
+    /**
+     * Handles notifications related to forum
+     *
+     * @param array $details
+     */
+    private function handleForum($details) {
```

Comment:
Instead of passing a `details` array here, make this function take explicitly named parameters